jerry-core/ecma/base/ecma-globals.h

```diff
@@ -1541,6 +1541,27 @@ typedef struct
  */
 #define ECMA_SYMBOL_HASH_SHIFT 2
 
+#if (JERRY_STACK_LIMIT != 0)
+/**
+ * Check the current stack usage by calculating the difference from the initial stack base.
+ * If the limit is reached a RangeError is raised.
+ */
+#define ECMA_CHECK_STACK_USAGE() \
+do \
+{ \
+  volatile int __sp; \
```

Comment:
My concern is that stack space will be allocated for this variable, and it will remain on the stack during the entire call chain, potentially adding up. I would consider moving this to a noinline function, and get the address from there, so that it gets freed immediately. I'd rather have a slight performance penalty than a slight increase in stack usage. What do you think?

Reply:
Sure, I can agree with this idea.